theSrc/scripts/PlotlyChartElements.js

```diff
@@ -0,0 +1,253 @@
+            showticklabels: config.showXAxis,
+            tickcolor: '#eee',
+            ticklen: 5,
+            tickfont: {
+                family: config.xAxisFontFamily,
+                color: config.xAxisFontColor,
+                size: config.xAxisFontSize
+            },
+            linecolor: config.plotBorderShow ? config.plotBorderColor : 'transparent',
+            linewidth: 1,
+            scaleratio: 1,
+            scaleanchor: config.fixedAspectRatio ? 'y' : null,
+            // draw zero line separately to ensure it sit on top layer
+            zeroline: false,
+            automargin: true,
+            range: [config.xBoundsMinimum, config.xBoundsMaximum],
+            dtick: parseTickDistance(config.xBoundsUnitsMajor),
+            tickprefix: config.xPrefix,
+            ticksuffix: config.xSuffix,
+            layer: 'below traces'
+         },
+        yaxis: {
+            title: {
+                text: config.yTitle,
+                font: {
+                    family: config.yTitleFontFamily,
+                    color: config.yTitleFontColor,
+                    size: config.yTitleFontSize
+                }
+            },
+            color: '#444',
+            showgrid: config.grid,
+            gridcolor: '#eee',
+            showticklabels: config.showYAxis,
+            tickcolor: '#eee',
+            ticklen: 5,
+            tickfont: {
+                family: config.yAxisFontFamily,
+                color: config.yAxisFontColor,
+                size: config.yAxisFontSize
+            },
+            linecolor: config.plotBorderShow ? config.plotBorderColor : 'transparent',
+            linewidth: 1,
+            scaleratio: 1,
+            scaleanchor: config.fixedAspectRatio ? 'x' : null,
+            // draw zero line separately to ensure it sit on top layer
+            zeroline: false,
+            range: [config.yBoundsMinimum, config.yBoundsMaximum],
+            dtick: parseTickDistance(config.yBoundsUnitsMajor),
+            tickprefix: config.yPrefix,
+            ticksuffix: config.ySuffix,
+            automargin: true,
+            layer: 'below traces'
+        },
+        title: {
+            text: config.title,
+            font: {
+                family: config.titleFontFamily,
+                color: config.titleFontColor,
+                size: config.titleFontSize
+            },
+            xref: 'paper',
+            automargin: false // setting this to true stuffs up alignment
+        },
+        showlegend: config.legendShow,
+        legend: {
+            font: {
+                family: config.legendFontFamily,
+                color: config.legendFontColor,
+                size: config.legendFontSize
+            },
+            itemsizing: 'constant',
+            yref: 'paper',
+            y: 0.5,
+            yanchor: 'middle',
+        },
+        margin: {
+            t: 20 + (config.title.length ? config.titleFontSize * 1.25 : 0),
```

Comment:
I assume this is translated over from R code?

Reply:
I've changed this in [1a634a8](https://github.com/Displayr/rhtmlCombinedScatter/pull/7/commits/1a634a8d45572edebe6a5a52447ba3688e8456f0) so that the margin can be set from buildConfig. Not sure if the plotly defaults are too wide, but we can leave this for later I think